### PR TITLE
frontend: Add system information

### LINKED ISSUE #577: core: frontend: Add system information page
Using linux2rest API information

### PR DESCRIPTION
**Pull request by patrickelectric**
Available in: patrickelectric/companion-core:system_information
Peek 27-01-2022 15-22

Fix #577

### CODE REVIEW

File: core/frontend/src/types/system-information/info.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/** Base structure that provides information about OS */
```
Comment (rafaellehmkuhl) [inline core/frontend/src/types/system-information/info.ts:1]: Are Info and KernelMessage the same?

File: core/frontend/src/types/system-information/system.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    available_space_B: number,
+    filesystem_type: string,
+    mount_point: string,
+    name: string,
+    total_space_B: number,
+    type: string
+}
+
+/** Base structure that provides basic OS/computer information */
+export interface Info {
```
Comment (rafaellehmkuhl) [inline core/frontend/src/types/system-information/system.ts:36]: Will this conflict with the other Info interface?

File: core/frontend/src/components/system-information/AboutThisSystem.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        { os: 'ubuntu', icon: 'mdi-ubuntu' },
+        { os: 'apple', icon: 'mdi-apple' },
+        { os: 'windows', icon: 'mdi-microsoft-windows' },
+      ]
+
+      if (info) {
+        for (const item of map) {
+          if (info.system_name.toLowerCase().includes(item.os)) {
+            return item.icon
+          }
```
Comment (rafaellehmkuhl) [inline core/frontend/src/components/system-information/AboutThisSystem.vue:80]: `return map.find(item => info.system_name.toLowerCase().includes(item.os)).icon`

File: core/frontend/src/views/SystemInformationView.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+      <v-col
+        v-if="true"
+        md="9"
+      >
+        <processes v-if="page_selected == 'process'" />
+        <system-condition v-if="page_selected == 'system_condition'" />
+        <network v-if="page_selected == 'network'" />
+        <about-this-system v-if="page_selected == 'about'" />
+      </v-col>
```
Comment (rafaellehmkuhl) [inline core/frontend/src/views/SystemInformationView.vue:51]: Do we want to consider browser history (being able to go back and forth) on those submenus? If so, we should implement routes here. If not, it is ok as is.

File: core/frontend/src/types/system-information/system.ts (hunk 2 above)
Comment (patrickelectric) [reply core/frontend/src/types/system-information/system.ts:36]: Only if you include it, but you can also include with a different name.
`import { info as SystemInfo } ...`

File: core/frontend/src/types/system-information/info.ts (hunk 1 above)
Comment (patrickelectric) [reply core/frontend/src/types/system-information/info.ts:1]: No, `Info` is OS specific, `KernelMessage` is for each kernel message. 
As in the documentation:
```
/** Base structure that provides information about kernel message */
```

File: core/frontend/src/views/SystemInformationView.vue (hunk 4 above)
Comment (patrickelectric) [reply core/frontend/src/views/SystemInformationView.vue:51]: I think that is not necessary.

File: core/frontend/src/components/system-information/AboutThisSystem.vue (hunk 3 above)
Comment (patrickelectric) [reply core/frontend/src/components/system-information/AboutThisSystem.vue:80]: Good catch!

Comment (patrickelectric) [reply core/frontend/src/components/system-information/AboutThisSystem.vue]: Done.

File: core/frontend/src/components/system-information/NetworkCard.vue [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    <v-list-item>
+      <v-list-item-icon>
+        <v-icon>
+          mdi-badge-account-horizontal
+        </v-icon>
+      </v-list-item-icon>
+      <v-list-item-subtitle>MAC:</v-list-item-subtitle>
+      <v-list-item-subtitle
+        class="text-right"
+        :style="show_mac ? '' : 'filter:blur(0.5em)'"
```
Comment (rafaellehmkuhl) [inline core/frontend/src/components/system-information/NetworkCard.vue:50]: Suggestion: using `cursor: pointer` css here on hover (`class:hover{}`) to change the mouse cursor, so user understand this is clickable (when blured). Right now, it's a text-selection cursor.

Not necessary, but pretty.

File: core/frontend/src/types/system-information/info.ts (hunk 1 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/types/system-information/info.ts:1]: But are they any different? They seem to have the same parameters.

Comment (patrickelectric) [reply core/frontend/src/types/system-information/info.ts:1]: Oh, this is a development file. I'll remove.

**Review by rafaellehmkuhl — APPROVED** [review]
LGTM

Great PR!

**Pull request merged**